src/Cutscenes.ts

```diff
@@ -1395,6 +1395,112 @@ export class Cutscenes<TEightBittr extends FullScreenPokemon> extends EightBittr
+                const left: number = direction === -1 ? iceBall.left : iceBall.right - 3 * this.EightBitter.unitsize;
+                const top: number =  iceBall.bottom - 3 * this.EightBitter.unitsize;
+                const line: IThing = this.EightBitter.things.add("IceBeamLine", left, top);
+
+                this.EightBitter.TimeHandler.addEvent(
+                    (): void => this.EightBitter.physics.shiftHoriz(iceBall, differenceX));
+                this.EightBitter.TimeHandler.addEvent(
+                    (): void => this.EightBitter.physics.shiftVert(iceBall, differenceY));
+                if (direction === 1) {
+                    this.EightBitter.graphics.flipHoriz(line);
+                }
+
+                lines.push(line);
+            },
+            1,
+            32);
+
+        this.EightBitter.TimeHandler.addEvent(
+            (): void => {
+                this.EightBitter.physics.killNormal(iceBall);
+                for (const line of lines) {
+                    this.EightBitter.physics.killNormal(line);
+                }
+                this.EightBitter.things.add(icicles[0], defender.left, defender.bottom - 4.5 * this.EightBitter.unitsize);
+
+                this.EightBitter.TimeHandler.addEvent(
+                    (): void => {
+                        this.EightBitter.physics.killNormal(icicles[0]);
+                        this.EightBitter.things.add(icicles[1], defender.left, defender.bottom - 8 * this.EightBitter.unitsize);
+
+                        this.EightBitter.TimeHandler.addEvent(
+                            (): void => {
+                                this.EightBitter.physics.killNormal(icicles[1]);
+                                this.EightBitter.things.add(icicles[2], defender.left, defender.bottom - 12.5 * this.EightBitter.unitsize);
+
+                                this.EightBitter.TimeHandler.addEvent(
+                                    (): void => {
+                                        this.EightBitter.physics.killNormal(icicles[2]);
+                                        this.EightBitter.things.add(
+                                            icicles[3],
+                                            defender.left,
+                                            defender.bottom - 20.5 * this.EightBitter.unitsize);
+
+                                        this.EightBitter.TimeHandler.addEvent(
+                                            (): void => {
+                                                this.EightBitter.physics.killNormal(icicles[3]);
+                                                this.EightBitter.animations.animateFlicker(defender, 14, 5, args.callback);
+                                            },
+                                            9 * 1);
+                                    },
+                                    9 * 1);
+                            },
+                            9 * 1);
+                    },
+                    9 * 1);
```

Comment:
This is something called the "pyramid of doom". You don't want that. Instead, just have a bunch of separate events all scheduled for different delays.